.github/workflows/nightly-upstream-snapshot-build.yml

```diff
@@ -115,24 +99,38 @@ jobs:
       test_ref: 'terraform'
       caller-workflow-name: 'nightly-upstream-snapshot-build'
 
-  # E2E tests where SampleApp has Java Agent
```

Comment:
Removed this Test as it builds the sample app with the released upstream dependencies and not with the snapshot one. I think providing this test here, requires much more work to ensure sample apps are build with java agent that have upstream snapshot versions.

Reply:
Was this job recently added to the nightly build workflow? I am having a bit of trouble tracing the git blame back for it.

Reply:
Yes I added it recently in [this](https://github.com/aws-observability/aws-otel-java-instrumentation/pull/627) PR. and I realized it was a mistake, so removing it with this PR. 
I think this testing should be part of nightly build, but putting it in `nightlybuild` workflows require some updates in the repo to facilitate this.